src/ripple/app/tx/impl/SetHook.cpp

```diff
@@ -922,7 +926,7 @@ SetHook::destroyNamespace(
         sleAccount->setFieldU32(sfHookStateCount, stateCount);
 
     if (ctx.rules.enabled(fixNSDelete))
-        adjustOwnerCount(view, sleAccount, -toDelete.size(), ctx.j);
+        adjustOwnerCount(view, sleAccount, -toDeleteOwnerCount, ctx.j);
```

Comment:
TODO: needs tests that OwnerCount decreases depending on HookState size when NS deleted.